<launch>
  <arg name="base"      value="$(optenv TURTLEBOT_BASE kobuki)"/> <!-- create, roomba -->
  <arg name="gui" default="true"/>
  <arg name="stacks"    value="$(optenv TURTLEBOT_STACKS hexagons)"/>  <!-- circles, hexagons --> 
  <arg name="3d_sensor" value="$(optenv TURTLEBOT_3D_SENSOR kinect)"/>  <!-- kinect, asus_xtion_pro --> 
    
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="use_sim_time" value="true"/>
    <arg name="debug" value="false"/>
    <arg name="gui" value="$(arg gui)" />    
    <arg name="world_name" value="$(find deu_car)/world/car_test_map.world"/>
  </include>


  <include file="$(find deu_car)/launch/kobuki2.launch.xml">
    <arg name="base" value="kobuki"/>
    <arg name="stacks" value="hexagons"/>
    <arg name="3d_sensor" value="kinect"/>
  </include> 

  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
      <param name="publish_frequency" type="double" value="30.0" />
  </node>
  
  <!-- Fake laser -->
  <node pkg="nodelet" type="nodelet" name="laserscan_nodelet_manager" args="manager"/>
  <node pkg="nodelet" type="nodelet" name="depthimage_to_laserscan"
        args="load depthimage_to_laserscan/DepthImageToLaserScanNodelet laserscan_nodelet_manager">
    <param name="scan_height" value="10"/>
    <param name="output_frame_id" value="/camera_depth_frame"/>
    <param name="range_min" value="0.45"/>
    <remap from="image" to="/camera/depth/image_raw"/>
    <remap from="scan" to="/scan"/>
  </node>

  <node pkg="deu_car" type="obstacle_spawn.py" name="obstacle_cube_01" args="$(find deu_car)/world/models/" />
  <node pkg="deu_car" type="drive.py" name="drive" args="1" output="screen"/>
  <node pkg="deu_car" type="follow_yellow_right.py" name="follow_yellow_right" output="screen"/>
  <node pkg="deu_car" type="follow_parking.py" name="follow_parking"/>
  <node pkg="deu_car" type="follow_yellow.py" name="follow_yellow" output="screen"/>
  <arg name="node_start_delay" default="3.0" />  
  <node pkg="deu_car" type="follow_white.py" name="follow_white" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' " output="screen"/>
  <node pkg="deu_car" type="follow_bar.py" name="follow_bar"/>
  <node pkg="deu_car" type="blocking_bar_control.sh" name="blocking_bar_01"  />
</launch>
